packages/perseus/src/validation.types.ts

```diff
@@ -278,11 +280,73 @@ export type UserInput =
     | PerseusSorterUserInput
     | PerseusTableUserInput;
 
-export type UserInputMap = {[widgetId: string]: UserInput | UserInputMap};
```

Comment:
Wasn't this here because some of the utility functions were recursive and might have nesting going on? Is that not the case anymore? May not have understood how it was used originally, so just checking :)

Reply:
The only widget that did that was the `group`.  I added a proper type for the `group`'s user input (which is `export type PerseusGroupUserInput = UserInputMap;`). So it makes this type a bit more accurate and the code that uses the `UserInput` easier to work with. I have a future PR almost ready where this type will be further improved.